internal/vm/setup.go

```diff
@@ -61,6 +61,15 @@ _G["process_auto_fields"] = function(plugin)
   return json.encode(p)
 end
 
+_G["validate_plugin_schema"] = function(plugin_schema_string)
+  local plugin_schema = loadstring(plugin_schema_string)()
+  local ok, err_t = MetaSchema.MetaSubSchema:validate(plugin_schema)
```

Comment:
This may not be enough. Can this function load and unload the schema?

Reply:
I am not tracking, can you provide more context as to why we would want to do this; are you not confident with the Lua source MetaSchema.MetaSubSchema:validate() function?

If we add the load functionality here then we are not doing anything different than the `LoadSchema` essentially opening the door to overriding the value in `Plugins`. We could perform the following for completeness if that is the concern:
1. Validate the schema and retrieve the plugin name
2. Ensure the plugin name does not exists in the `Plugins` instance
3. Create a new instance (e.g. `local temp_plugins = assert(Entity.new(plugins_definition))`
4. Load the schema and asses the result
5. Return the plugin name to the caller.

Here the local instance will not overwrite the global `Plugins` instance in the event it was loaded; a layer of isolation.

Reply:
Added all the validation and loading checks in [7348683](https://github.com/Kong/goks/pull/15/commits/7348683cbc1474f95e23847cb4fb5441c5f0e9a4)